cns/wireserver/net.go

```diff
@@ -79,3 +124,9 @@ func calculateGatewayIP(cidr string) (net.IP, error) {
 	}
 	return gw, nil
 }
+
+func macAddressesEqual(macAddress1, macAddress2 string) bool {
```

Comment:
Rather than undifferentiated strings, I'd just make:

```go
type MACAddress string

func (m MACAddress) Equal(other MACAddress) bool {
       // ...everything in here
}
```

MAC addresses are things that can be compared, validated, dissected, etc., so they deserve a type.

Reply:
@timraymond I agree with this if there is more use-case for this type than just for string comparison here..the main caller of this method comes from ipam.go(L160) where it passes a field  `ipconfigResponse.PodIPInfo[i].MacAddress` so I don't want to convert this to a new type MACAddress & back to strings inside for comparison. If this was used for more methods or part of a struct I could add it, but it seems more convert-to&back just for one use.

Reply:
we should absolutely not be using strings here, there is a stdlib MAC type: https://pkg.go.dev/net#HardwareAddr

Reply:
this refactor means it would have to be refactored at the caller level too, should we change the type for MacAddress field in PodIpInfo? - 
![image](https://github.com/Azure/azure-container-networking/assets/26335310/0cdeab6c-ccbf-4846-974f-13276cd1d4d7)

Reply:
let me try to explore this library to compare the two macs instead of this function

Reply:
I tried to use the package& parse the mac string but unfortunately the format in which wireserver api returns the MACAddress is not parsed by this package as well - 

@rbtr do you suggest any other package that could format this for comparison

Reply:
this was recently raised actually https://github.com/golang/go/issues/66682
Generally, we should be converting this sort of thing to a type at the boundary of its usage so that we can use the types without worrying about format differences and special cases every time we use it.
If the res.Interface.MacAddress is of type net.HardwareAddr already, you don't have to parse it here at all 😉 We can consolidate any special handling of MAC parsing in to a single place and simply compare HardwareAddr == HardwareAddr here